functions/graphql/schema/badge.js

```diff
@@ -96,15 +103,20 @@ module.exports.schema = gql`
 // deep merged with the other resolvers.
 module.exports.resolver = {
   Query: {
+    // Anyone should be able to see any badge
     badges: async (rootQuery, { type, centerId }, context) => {
       return Badge.getBadges(type, centerId);
     },
+    // Anyone should be able to see any badge
     badge: (rootQuery, { id }, context) => {
       return Badge.getBadge(id);
     }
   },
   Mutation: {
+    // You should only be able to create a badge for a center that you own
     badgeCreate: async (rootQuery, { badge, centerId }, context) => {
+      console.log(context.user);
```

Comment:
This is now broken. It won't ever create a badge properly. Also, if we put the `@auth(requires:[director])` directive on the mutation schema definition, it will block it, since we have the centerId in the arguments of the mutation.